src/dune_lang/decoder.ml

```diff
@@ -377,6 +377,12 @@ let int =
       | x -> Ok x
       | exception _ -> Result.Error ())
 
+let float =
+  basic "Float" (fun s ->
+      match float_of_string s with
```

Comment:
Could you add `Float.of_string` and use that? This function should return `float option`.

Reply:
I don't mind of course, but I did respect the current convention I think. The `int` function just above reads

```
let int =
  basic "Integer" (fun s ->
      match int_of_string s with
      | x -> Ok x
      | exception _ -> Result.Error ())
```
I can change it but that would actually be inconsistent, maybe we should either keep it that way or do a pass on the whole encoder module ?